spring-cloud-stream/src/test/java/org/springframework/cloud/stream/function/SourceToFunctionsSupportTests.java

```diff
@@ -0,0 +1,226 @@
+						TestChannelBinderConfiguration.getCompleteConfiguration(SupplierConfiguration.class))
+						.web(WebApplicationType.NONE)
+						.run("--spring.cloud.stream.function.name=number|concatWithSelf|mmultiplyByTwo",
+								"--spring.jmx.enabled=false");
+
+		OutputDestination target = context.getBean(OutputDestination.class);
+		assertThat(target.receive(10000).getPayload())
+				.isEqualTo("22".getBytes(StandardCharsets.UTF_8));
+		assertThat(target.receive(10000).getPayload())
+				.isEqualTo("44".getBytes(StandardCharsets.UTF_8));
+		assertThat(target.receive(10000).getPayload())
+				.isEqualTo("66".getBytes(StandardCharsets.UTF_8));
+		//etc
+	}
+
+	@EnableAutoConfiguration
+	@Import(ProvidedMessageSourceConfiguration.class)
+	public static class SupplierConfiguration {
+		AtomicInteger counter = new AtomicInteger();
+		@Bean
+		public Supplier<String> number() {
+			return () -> String.valueOf(counter.incrementAndGet());
+		}
+
+		@Bean
+		public Function<String, String> concatWithSelf() {
+			return x -> x + x;
+		}
+
+		@Bean
+		public Function<Flux<String>, Flux<String>> mmultiplyByTwo() {
+			return x -> x.map(i -> String.valueOf(Integer.valueOf(i) * 2));
+		}
+	}
+
+	@EnableAutoConfiguration
+	@Import(ExistingMessageSourceConfiguration.class)
+	public static class FunctionsConfiguration {
+
+		@Bean
+		public Function<String, String> toUpperCase() {
+			return String::toUpperCase;
+		}
+
+		@Bean
+		public Function<String, String> concatWithSelf() {
+			return x -> x + ":" + x;
+		}
+	}
+
+	/**
+	 * This configuration essentially emulates our existing app-starters for Sources
+	 * and essentially demonstrates how a function(s) could be applied to an existing
+	 * source via {@link IntegrationFlowFunctionSupport} class.
+	 */
+	@EnableBinding(Source.class)
+	public static class ExistingMessageSourceConfiguration {
+
+		@Autowired
+		private Source source;
+
+		@Bean
+		public IntegrationFlow messageSourceFlow(IntegrationFlowFunctionSupport functionSupport) {
+			Supplier<Message<String>> messageSource = () -> MessageBuilder.withPayload("hello function")
+					.setHeader(MessageHeaders.CONTENT_TYPE, MimeTypeUtils.TEXT_PLAIN).build();
+
+			IntegrationFlowBuilder flowBuilder = functionSupport.integrationFlowFromProvidedSupplier(messageSource);
+
+			if (!functionSupport.andThenFunction(flowBuilder, source.output())) {
+				flowBuilder = flowBuilder.channel(source.output());
+			}
+
+			return flowBuilder.get();
+		}
+
+	}
+
+	@EnableBinding(Source.class)
+	public static class ProvidedMessageSourceConfiguration {
+
+		@Autowired
+		private Source source;
+
+		@Autowired
+		private FunctionProperties functionProperties;
+
+		@Bean
+		public IntegrationFlow messageSourceFlow(IntegrationFlowFunctionSupport functionSupport) {
+			Assert.hasText(functionProperties.getName(), "Supplier name must be provided");
+
+			IntegrationFlowBuilder flowBuilder = functionSupport
+					.integrationFlowFromNamedSupplier()
```

Comment:
Well, I feel a bit uncomfortable with this API: or should request a supplier name from end-user.
Or just make it `Supplier`-based and end-users to inject resolved upfront name-based suppliers.
At least that is how we deal with auto-configuration.

On the other hand the logic in the `IntegrationFlowFunctionSupport` should not be aware of auto-configurations, e.g. the same `FunctionProperties`. 

Just thoughts from my feeling...

Reply:
Ok, let me try to clarify. . .
Every bean (as `@Bean`) of type `Supplier`, `Function` and/or `Consumer` will be loaded to `FunctionCatalog` by SCF. 
What we're doing here is simply specifying which one to use via `spring.cloud.stream.function.name` property.
So, it's not really comparable with auto-configuration concepts.

Does this clarify?

Reply:
What you mean sounds like a `defaultFunction` (or name) in the `IntegrationFlowFunctionSupport`, since you indeed goes to the catalog for lookup. That's why I don't see reason to be tied with the implicit name from some hard-coded property. Would be really better to configure this `IntegrationFlowFunctionSupport` bean appropriately leaving a door for end-users to customize.

Does it make sense?

Reply:
Unless I am missing something, I am not sure I follow the _". . .configure this IntegrationFlowFunctionSupport bean appropriately. . ."_

Just to clarify a bit:
Doing it via property ensures that I can
* change the actual function that I want to use without rebuilding the app
* the name can include composition (i.e., `foo|bar|baz`) which essentially extends on the previous point
* the long term goal is to also consider making this property optional if there is only one applicable function in the catalog

Reply:
Well, I don't mind about that configuration property. Only what is my concern that it is not an `IntegrationFlowFunctionSupport` responsibility. 
you need to resolve a property to the name in the `FunctionConfiguration` and allow end-user to override it any possible way without any dependency on the `FunctionProperties`. 
That is what my concern about mixing auto-configuration goodies in this support component...

Well, feel free to reject my opinion and leave it as is.
Just let me know what is the final decision do not come back to this subject anymore! 😄